src/metabase/api/embed.clj

```diff
@@ -172,21 +173,23 @@
       add-implicit-card-parameters
       :parameters))
 
-(defn- resolve-dashboard-parameters
-  "Returns parameters for a card on a dashboard with `:target` resolved via `:parameter_mappings`."
-  [dashboard-id dashcard-id card-id]
-  (let [param-id->param (u/key-by :id (for [param (db/select-one-field :parameters Dashboard :id dashboard-id)]
-                                        (update param :type keyword)))]
-    ;; throw a 404 if there's no matching DashboardCard so people can't get info about other Cards that aren't in this
-    ;; Dashboard. We don't need to check that `card-id` matches the DashboardCard because we might be trying to get
-    ;; param info for a series belonging to this dashcard (`card-id` might be for a series)
-    (for [param-mapping (api/check-404 (db/select-one-field :parameter_mappings DashboardCard
-                                         :id           dashcard-id
-                                         :dashboard_id dashboard-id))
-          :when         (= (:card_id param-mapping) card-id)
-          :let          [param (get param-id->param (:parameter_id param-mapping))]
-          :when         param]
-      (assoc param :target (:target param-mapping)))))
+(s/defn ^:private resolve-dashboard-parameters :- [dashboard-api/ParameterWithID]
+  "Given a `dashboard-id` and parameters map in the format `slug->value`, return a sequence of parameters with `:id`s
+  that can be passed to various functions in the `metabase.api.dashboard` namespace such as
+  `metabase.api.dashboard/run-query-for-dashcard-async`."
+  [dashboard-id :- su/IntGreaterThanZero
+   slug->value  :- {s/Any s/Any}]
+  (let [parameters (db/select-one-field :parameters Dashboard :id dashboard-id)
+        slug->id   (into {} (map (juxt :slug :id)) parameters)]
+    (vec (for [[slug value] slug->value
+               :let         [slug (u/qualified-name slug)]]
+           {:slug  slug
+            :id    (or (get slug->id slug)
+                       (throw (ex-info (tru "No matching parameter with slug {0}. Found: {1}" (pr-str slug) (pr-str (keys slug->id)))
+                                       {:status-code          400
+                                        :slug                 slug
+                                        :dashboard-parameters parameters})))
+            :value value}))))
```

Comment:
This code is simplified a bit now and just maps the parameter slug to the correct parameter; the new stuff in `metabase.api.dashboard` can handle the DashCard parameter mapping stuff now.